mock/src/builders/tx.rs

```diff
@@ -0,0 +1,68 @@
+use miden_objects::accounts::AccountId;
+use miden_objects::notes::{NoteEnvelope, Nullifier};
+use miden_objects::transaction::{
+    InputNotes, OutputNotes, ProvenTransaction, ToEnvelope, ToNullifier,
+};
+use miden_objects::vm::ExecutionProof;
+use miden_objects::Digest;
+use miden_prover::{HashFunction, StarkProof};
+
+/// Builder for an `ProvenTransaction`, the builder allows for a fluent API to construct an account.
+/// Each transaction needs a unique builder.
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct ProvenTransactionBuilder {
```

Comment:
Is this meant to replace [`MockProvenTxBuilder`](https://github.com/0xPolygonMiden/miden-node/blob/cb58a159d66afb52153161519340b688c66ba4bc/block-producer/src/test_utils/proven_tx.rs#L35) in `miden-node`?

Reply:
If it is possible, I would think it is a good idea. The rationale is that having fewer utilities doing the same is best, and the utilities in `miden-base` can be used on the other projects, like `miden-client`.